### ISSUE TITLE
Info manual improvements

### ISSUE THREAD

**Issue by DSMasterson** [alphapapa/org-super-agenda#67]
I'm starting to look at org-super-agenda, but the documentation is particularly difficult to follow.  Installing it via MELPA gets it installed properly (I think) with an Info file, but the Info file does not look at all good.  For instance, the pictures are totally gone when viewed in Emacs Info.  I would suggest redoing your Info file from scratch -- perhaps as an Org file exported into Texinfo format which you can then generate into the Info file (if you're really industrious, you could tangle the Org file into your Elisp for Super Agenda).  The docs need more examples of what a typical setup of an Agenda would look like in Elisp and then how you'd modify the Elisp to use Super Agenda (I'm still figuring out all the Org variables).

Super Agenda looks promising with good documentation...

**Comment by alphapapa** [comment]
Well, after over 18 months and 3,200 downloads, most users seem to figure it out, so I don't think the docs are that bad.  If you don't already understand how Org Agenda works, and you don't know some elisp basics, then you may not easily understand how it works, but that's out-of-scope for this project.  There are excellent resources available for learning about both of those.

Most of the questions I get are from people who don't carefully read the existing documentation.  e.g. many people skip step 1, and many other people skip the explicit note on step 2.  If people rush through the step-by-step documentation instead of reading it carefully, I can't fix that by writing more documentation.

The pictures are not important for the info manual, they are merely pretty pictures to show potential users what it looks like before they download it.

The info manual is already exported from the Org file, just like other projects that do that, like Magit.  If you have any specific recommendations for options to set to improve the exported info manual, please post them.

I am not converting this project into a giant Org file to be tangled with org-babel.  That would be a lot of work for no benefit.

There is already a dedicated examples.org file with 14 examples of how to use different features.

So, in conclusion, while the docs aren't perfect, they are already extensive, and 99% of the questions I get are from people who haven't actually read them.  There's nothing I can do about that.

If you have specific suggestions for improvements, those would be welcome.

**Comment by DSMasterson** [comment]
Okay.  I apologize.  I posed that incorrectly.  I'm not trying to be difficult -- I think Super Agenda has potential and I might want to help.  I have been around Emacs off and on for over 35 years (yes, since it was written in TECO), so I'm familiar with a lot of the structure of Emacs and can read a lot of Elisp, but I'm not an expert or an Elisp programmer.  The problems that I see:

1. I was looking at it from just what was delivered in the MELPA package and from a users perspective.
2. I don't see your name in the Info file.  I dug it out of the comments in org-super-agenda.el. 
3. You are asking for contributions, but you don't explain how to contribute in the Info file (mention github, etc.) which I also picked up in your .el file.
4.  Your example was a good starting point, but I'm still getting my feet wet with Org agendas, so it still hasn't clicked about all the ins and outs of connecting it to base Org.  Does the Super Agenda apply to all agendas (standard and custom) or do you control it and, if so, how?
5.  I don't see your Texinfo file that you used to produce your Info file in your GitHub master branch.  Did you use a Texinfo file to produce it?  Could you add that to the release?
6.  Examples.org does not appear (to me) to be in the MELPA release.  It is not in the installation on my machine, but I can see it in the GitHub repository.  I also don't see it mentioned in the Info file, so the normal user would not know anything about it.  Am I missing something?

**Comment by alphapapa** [comment]
> Okay. I apologize. I posed that incorrectly. I'm not trying to be difficult -- I think Super Agenda has potential and I might want to help. I have been around Emacs off and on for over 35 years (yes, since it was written in TECO), so I'm familiar with a lot of the structure of Emacs and can read a lot of Elisp, but I'm not an expert or an Elisp programmer.

No problem.  I'm glad to have veteran Emacs users like yourself take an interest and lend their perspective and expertise.

> I don't see your name in the Info file. I dug it out of the comments in org-super-agenda.el.

That might be a good thing to add to the info file, although it's present in the source code, and most users will probably refer directly to the GitHub repo.  The info file is intended as a quick, in-Emacs reference rather than as a comprehensive manual, but I'm not opposed to improving it.

> You are asking for contributions, but you don't explain how to contribute in the Info file (mention github, etc.) which I also picked up in your .el file.

Well, the GitHub repo is linked in the source code, in the package headers, the package description available through MELPA and with `C-h P`, and in the customization group.  Those are the standard ways AFAIK.  The URL should probably be present in the Info manual somewhere, but as before, it's not intended as a comprehensive, GNU-style manual.  But improvements are welcome.  :)

> Your example was a good starting point, but I'm still getting my feet wet with Org agendas, so it still hasn't clicked about all the ins and outs of connecting it to base Org. Does the Super Agenda apply to all agendas (standard and custom) or do you control it and, if so, how?

Whenever `org-super-agenda-mode` is active, the grouping in `org-super-agenda-groups` is applied to all Org agenda-style commands, like `org-agenda`, `org-todo-list`, etc.  This is mentioned in the readme, but not very clearly or in the best place, so that should be improved.

> I don't see your Texinfo file that you used to produce your Info file in your GitHub master branch. Did you use a Texinfo file to produce it? Could you add that to the release?

The intermediate `texi` file  is generated by `org-texinfo-export-to-info` and then deleted in the `after-save-hook` in the readme:
```
after-save-hook: (lambda nil (when (and (require 'ox-texinfo nil t) (org-texinfo-export-to-info)) (delete-file "README.texi") (rename-file "README.info" "org-super-agenda.info" t)))
```
Since it's generated automatically, it isn't needed in the repo, and wouldn't be appropriate to modify directly, anyway.  

Probably the best example of exporting from Org to Info is found in Magit's documenation.  I copied some tricks from it, but more could be done.

> Examples.org does not appear (to me) to be in the MELPA release. It is not in the installation on my machine, but I can see it in the GitHub repository. I also don't see it mentioned in the Info file, so the normal user would not know anything about it. Am I missing something?

You're right, that file isn't in the info manual or the MELPA archive.  I'm not sure of the best way to include it, because I want to keep those examples separate from the readme file, and it wouldn't make sense to have them as a separate info file.  I think Org has an `#+INCLUDE:` directive for exporting, so maybe that would do it.

Thanks for your feedback.  I'll look into some of these issues when I get a chance.  Please let me know if I can help you with anything else.  If you're not already familiar with the Org Agenda, trying to figure out this at the same time can be overwhelming.

**Comment by DSMasterson** [comment]
Oh, that's right -- org-super-agenda.org gets generated into the Info file!  You should also include the Org file in the MELPA package to serve two purposes: a hacker (ie. me) might hack the file to produce better documentation and it might provide examples of Org and Super Agenda for new people.

#+INCLUDE should be the way to include examples.org in your info file, but also find a way to include the examples.org file in the package for people to refer to.  Just remember that most users of Emacs are not going to be GitHub users -- they'll just take the packages that are installed with their operating system and maybe use the package manager to update.  Ultimately, the extra .org files will be removed from the package when the code stabilizes.

**Comment by alphapapa** [comment]
> Oh, that's right -- org-super-agenda.org gets generated into the Info file! You should also include the Org file in the MELPA package to serve two purposes: a hacker (ie. me) might hack the file to produce better documentation and it might provide examples of Org and Super Agenda for new people.

Well, that's not how packages published on MELPA are generally intended to be packaged or worked on.  Org files aren't generally included because there's no standard, user-facing way to access them.  A user would have to manually look at the package's installed source directory to see what files are there, and since Org files aren't included in standard recipes, there's no expectation that such files would be there, so there's no reason for users to go looking for them--it's sort of a chicken-and-egg issue.  In contrast, info files are packaged by default, and package.el handles setting up the paths so `M-x info` can find them.

For hacking changes that would be contributed upstream, users are expected to clone the git repo and work on that.  Working on a detached copy would generally be a bad idea as it would easily diverge from upstream and require resolving conflicts to merge.

So I could add the Org file to the recipe, but 1) MELPA would have to approve the change to the recipe in order for MELPA users to receive the changes, which they might not do; and 2) as I explained, there's little reason to do so in the first place.

The same goes for the examples.org file.  If I can integrate it into the info manual, that would be good and appropriate, but packaging the Org file probably wouldn't be.

> Just remember that most users of Emacs are not going to be GitHub users -- they'll just take the packages that are installed with their operating system and maybe use the package manager to update. Ultimately, the extra .org files will be removed from the package when the code stabilizes.

Not sure exactly what you mean here.  Most users of this package install it from MELPA, and most of them aren't elisp hackers to begin with.  Those that are will have no problem finding the git repo.  I'm not aware of org-super-agenda being packaged for any OS distros (like some ELPA packages are for Debian), but if it were, the distro package maintainer could include whatever files he wanted; that wouldn't be up to me.

So maybe I'm misunderstanding you, but what I think you're suggesting would be more appropriate if this package were being proposed for inclusion with Org proper or Emacs itself, which I don't expect to ever happen.

**Comment by DSMasterson** [comment]
You're right.  I've been too out of touch with the development process of Emacs for far too long (pre-Github).  I think you're more correct than I was.  The best I could see would be just mention more about how to contribute to your project on Github in the Info file.  I don't mean explain how to use Github -- just where it is on Github and, perhaps, which branch(s) you expect people to commit to.  I'll have to play with Github more and see if I can find documentation on the differences between MELPA and ELPA.

**Comment by alphapapa** [comment]
I guess it wouldn't hurt to have the URL in the info file, but users who have installed it will have either visited the repo on GitHub already, or installed it from MELPA with package.el, which means `C-h P` will show the repo URL.  And it's visible with `M-x customize-group RET org-super-agenda RET`.  And there's always Google.  :)

ELPA is run by GNU/FSF, and all packages in it have copyright assigned to the FSF, and committers can commit to any package in ELPA (though I think most usually stick to the packages they authored).  MELPA is completely independent of the FSF and simply serves packages automatically built from recipes that point to the packages' repos (not necessarily on GitHub).
